services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/BandwidthSpec.groovy

```diff
@@ -162,4 +162,53 @@ class BandwidthSpec extends BaseSpecification {
         cleanup: "Delete created flow"
         flow?.id && northboundService.deleteFlow(flow.id)
     }
+
+    def "Able to update bandwidth to maximum link speed without using alternate links"() {
+        given: "Two active neighboring switches"
+        def switches = topology.getActiveSwitches()
+        def links = northboundService.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            links.any { link ->
+                def switchIds = link.path*.switchId
+                switchIds.contains(src.dpId) && switchIds.contains(dst.dpId)
+            }
+        }
+        assert srcSwitch && dstSwitch
+
+        when: "Create a flow with a valid small bandwidth"
+        def maximumBandwidth = 1000
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = maximumBandwidth
+        northboundService.addFlow(flow)
+
+        then: "Flow is really created and has 'Up' status"
+        Wrappers.wait(5) {
```

Comment:
Maybe it's worth adding assert that checks that flow is actually UP after the wait?

Reply:
Nevermind, looks like "then" clause itself should be enough.

Reply:
Yep.